File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
import sys
input = sys.stdin.readline

def f():
    if len(s) == m:
        print(*s)
        return
    else :
        for i in range(1,n+1):
            if i not in s:
                s.append(i)
                f()
                s.pop()


n, m = map(int,input().split())
s = []
f()

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [a, b] = input[0].split(" ").map(Number);
const tmp = []

function f() {
  if (tmp.length == b) {
    console.log(tmp.join(' '))
    return
  }

  else {
    for (let i = 1; i <= a; i++){
      if (!tmp.includes(i)) {
        tmp.push(i)
        f()
        tmp.pop()
      }
    }
  }
}

f()